### DISCUSSION TITLE
Currently failed and Total failed versus Currently banned and Total banned

### DISCUSSION THREAD

**Discussion by jaygooby** [fail2ban/fail2ban#3413] [Q&A] [answered]
When I run `sudo fail2ban-client status foo` you get an output like:

```
Status for the jail: foo
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     14
|  `- File list:        /var/log/access.log
`- Actions
   |- Currently banned: 8
   |- Total banned:     8
   `- Banned IP list:   5.181.86.99 135.125.21.38 81.19.136.5 45.227.253.70 78.128.113.58 179.60.147.173 45.227.254.17 78.128.112.86
```

What do **`Currently failed`** and **`Total failed`** measure?

**Comment by sebres** [comment, marked as answer]
Fail2ban has two internal lists managing tickets with failures (matches in filter) and bans.
Every fail ticket will left the fail-manager list if either the last failure of the IP/ID causes a ban reaching `maxretry` failures in `findtime` time window or if there are no failures from the IP/ID during the `findtime`.

`Currently failed` - number of tickets (IPs) with failures that did not caused a ban yet
`Total failed` - total number of failures of the jail since last start of fail2ban

`Currently failed: 0` means that at the moment there are no known tickets which are a) not banned yet, and b) not obsolete. With other words fail-manager list is empty. 
`Total failed:     14` means that since start of fail2ban filter of jail found 14 matches.

**Reply by jaygooby** [reply]
Ah, thanks! For that `foo` jail I've set `bantime = -1` so, I'd have thought that the `Total banned` and `Total failed` counts would be the same number, but I appreciate the detail in your answer

**Reply by sebres** [reply]
> the `Total banned` and `Total failed` counts would be the same number

Basically no. Firstly `Total banned` is a count of bans, `Total failed` is a count of **all** failures found by jail since its start. To get banned a ticket needs a `maxretry` failures, considering jails `findtime`. If `maxretry > 1` some of them may expire if no one failure occurs during the `findtime` window hereafter. Secondly there may be more failures from IP as expected for the ban (simply already logged several failing requests, that are initiated before the ban taken place), for instance even if `maxretry = 1` but IP made 2 failures simultaneously, it gets banned immediately after first of them gets found, but 2 failures will be counted in filter.
Also note that if you restart fail2ban or jail it would restore the active banned tickets, but not the failure list. 
With other words it is more complex.

As for persistent banning (`bantime = -1`) it is not recommended and ugly, because with growing the banned list simply bother fail2ban and net-filter subsystem with old addresses that probably never recurring anymore (let alone dynamic IPs)... it is almost a rudiment and retained only for backwards compatibility.
Instead, better use `bantime.increment = true` with some smaller initial `bantime` + something like `bantime.maxtime = 5w` - normally several weeks is really large enough to discourage all rude and recidive evildoers. And if not - they'd do maximal 10 attempts per year (what is far too little for a bruteforce attack). Or at most something like `bantime.maxtime = 180d` (or `bantime.maxtime = 1y`). I don't think half year or year is too small.
See [jail.conf](https://github.com/fail2ban/fail2ban/blob/bd6e7aeff029db1b52d169efb5f79ce1619ee8db/config/jail.conf#L47-L61) or its man-page for details.

**Reply by jaygooby** [reply]
> but IP made 2 failures simultaneously, it gets banned immediately after first of them gets found, but 2 failures will be counted in filter.
> Also note that if you restart fail2ban or jail it would restore the active banned tickets, but not the failure list.
With other words it is more complex.

Ah, OK. That's clear. I've definitely restarted after updating the jail regex. Thanks again for the detailed answer, and yep, several weeks makes sense for the `bantime`.
